cli.js

```diff
@@ -1,20 +1,128 @@
 #!/usr/bin/env node
-'use strict'
-const spawnSync = require('child_process').spawnSync
-const COMPILER_PATH = require('./index.js').compiler.COMPILER_PATH
-const result = spawnSync(
-  'java',
-  ['-jar', COMPILER_PATH].concat(process.argv.slice(2)),
-  {stdio: 'inherit'}
-)
-if (result.error) {
-  if (result.error.code === 'ENOENT') {
-    console.error('Could not find "java" in your PATH.')
-  } else {
-    console.error(result.error.message)
-  }
-  process.exit(1)
+'use strict';
+const fs = require('fs');
+const {getNativeImagePath, getFirstSupportedPlatform, parseCliFlags} = require('./lib/utils');
+
+const compilerFlags = parseCliFlags(process.argv.slice(2));
+let platform;
+if (compilerFlags.platform) {
+  platform = compilerFlags.platform;
+  delete compilerFlags.platform;
 } else {
-  process.exit(result.status)
+  platform = getFirstSupportedPlatform(['native', 'java', 'javascript']);
 }
 
+if (platform !== 'javascript') {
+  const Compiler = require('./lib/node/closure-compiler');
+  let args = process.argv.slice(2);
+  for (let i = 0; i < args.length; i++) {
+    if (/^--platform/.test(args[i])) {
+      let delCount = 1;
+      if (args[i].indexOf('=') < 0 && args.length > i + 1) {
+        delCount++;
+      }
+      args.splice(i, delCount);
+      break;
+    }
+  }
+
+  const compiler = new Compiler(args);
+
+  compiler.spawnOptions = {stdio: 'inherit'};
+  if (platform === 'native') {
+    compiler.JAR_PATH = null;
+    compiler.javaPath = getNativeImagePath();
+  }
+
+  compiler.run((exitCode) => {
+    process.exit(exitCode);
```

Comment:
Not in this CL, but I find things clearer if there's a main method that can set `process.exitCode`.  Linters do better when code uses `return` instead of functions that knowingly cease execution of the current function.